applications/RANSApplication/CMakeLists.txt

```diff
@@ -0,0 +1,87 @@
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+
+message("**** configuring KratosRANSApplication ****")
+
+# check for fluid dynamics application
+if(NOT (${FLUID_DYNAMICS_APPLICATION} MATCHES ON))
+  message(FATAL_ERROR "RANSApplication requires FluidDynamicsApplication, Please re-compile with -DFLUID_DYNAMICS_APPLICATION=ON flag.")
+endif(NOT (${FLUID_DYNAMICS_APPLICATION} MATCHES ON))
+
+################### PYBIND11
+include(pybind11Tools)
+
+include_directories( ${CMAKE_SOURCE_DIR}/kratos )
+include_directories( ${CMAKE_SOURCE_DIR}/applications/FluidDynamicsApplication )
+
+if((${MPI_NEEDED} MATCHES ON) AND (${TRILINOS_APPLICATION} MATCHES ON))
+  message("Enabling trilinos extension for KratosRANSApplication")
+  add_subdirectory(trilinos_extension)
+endif((${MPI_NEEDED} MATCHES ON) AND (${TRILINOS_APPLICATION} MATCHES ON) )
+
+## RANSApplication Core sources
+file(GLOB_RECURSE KRATOS_RANS_APPLICATION_CORE
+    ${CMAKE_CURRENT_SOURCE_DIR}/rans_application.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/rans_application_variables.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_conditions/*.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_elements/*.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_processes/*.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_strategies/*.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_utilities/*.cpp
+)
+
+file(GLOB_RECURSE KRATOS_RANS_APPLICATION_PYTHON_INTERFACE
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom_python/*.cpp
+)
+
+set(KRATOS_RANS_APPLICATION_SOURCES_MPI "")
+set(KRATOS_RANS_APPLICATION_EXTRA_LIBS "")
+if(${MPI_NEEDED} MATCHES ON )
+    include_directories( ${CMAKE_SOURCE_DIR}/applications/TrilinosApplication )
+    find_package(TRILINOS REQUIRED)
```

Comment:
You should only try to find trilinos if you don't have it yet, please check the trilinos extension in the FluidDynamics (@philbucher we need to find a better solution for this, probably calling find(trilinos) from the main CMakeLists if needed instead of from each application that needs it)

Reply:
I agree very much!
I would have looked into it after the new install mechanism is merged, which should happen in the next days
FYI @roigcarlo

Reply:
There will be a new `kratos_add_dependency(path_of_the_app)` macro which will let you add any dependency without having to care about if it has already been added or not. You can use that

Reply:
removed this... :)